build-system/tasks/build.js

```diff
@@ -39,11 +39,14 @@ const argv = require('minimist')(process.argv.slice(2));
  * Used by `gulp` and `gulp build`.
  *
  * @param {!Object} options
+ * @return {Promise}
  */
 async function runPreBuildSteps(options) {
-  await compileCss(options);
-  await compileJison();
-  await bootstrapThirdPartyFrames(options);
+  return Promise.all([
+    compileCss(options),
+    compileJison(),
```

Comment:
Shouldn't this be removed from here now that extensions that need it are are calling it independently?

Reply:
Currently `doBuild` will call `compileAllJs` before `buildAllExtensions`. And compileAllJs includes ww.js, which needs the parser. Ideally we `ww.js` could somehow know about its dependency on the parser but I can't think of a solution besides for special casing it :(

Reply:
There is precedent for special-casing specific extensions and doing extra stuff during `buildExtension`. (Search for `if (name`.)

https://github.com/ampproject/amphtml/blob/df83808bc81014fc23fe9efb58f04bc91e1ff702/build-system/tasks/extension-helpers.js#L420-L459

Since `ww.js` is built by `compileJs` in `helpers.js`, perhaps add a function that does a special-case extra step and conditionally invoke it at the end of `compileMinifiedJs` / `compileUnminfiedJs`?

Reply:
Since ww.js is not technically an extensions, this would need to live elsewhere. I'll add a special case though and see how icky it feels.

Reply:
Oh! Unless you mean to add `doBundle(ww.js)` as part of building amp-bind, which actually would let me put this within buildExtension!

Reply:
I did initially mean the icky special case, but if `amp-bind` is the only extension that uses `ww.js`, then I think your idea is better.